### DISCUSSION TITLE
Is it allowed to keep all functionality in just one default.html?

### DISCUSSION THREAD

**Discussion by webNEXT-development** [datenstrom/community#872] [Ask a question] [answered]
Hi all,
I am creating a new template for yellow. I was wondering why the layouts folder has different template *.html files like (default, footer, navigation, etc), what is the reason for doing this? 

My question is "Is it allowed to keep all functionality in just one default.html".

The new template contains the files:
default-website.html
default-blog.html
default-wiki.html

Regards,
Marcel

**Comment by GiovanniSalmeri** [comment]
Hello Marcel!

First of all: could you please edit the Title of your question and write there the text of the question? (i.e. "Is it allowed to keep all functionality in just one default.html?"). You should use the button Edit at the top right corner. Otherwise it is not clear from the title what you are asking for and the whole discussion risks being flagged and deleted.

Regarding your question: I have never extensively worked with templates, but I imagine that having different files (default.html, header.html etc.) helps in keeping things in order and makes possible to reuse single fragments of template. But I do not see anything that prevents writing a default.html which contains all the required functionality.

**Reply by annaesvensson** [reply]
In the standard installation the HTML code is assembled from [multiple files](https://github.com/datenstrom/yellow/tree/main/system/layouts). One reason for multiple files is so that you can re-use HTML code in your layouts. For example write `<?php $this->yellow->layout("navigation") ?>` where you want to have a navigation. Another reason for multiple files is so that you can customise HTML code used in multiple layouts. For example edit the file `system/layouts/navgation.html` and all layouts will use the same navigation. Now you have the ability to assemble HTML code from smaller building blocks:

``` php
<?php $this->yellow->layout("header") ?>
<div class="content">
<div class="main" role="main">
<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
<?php echo $this->yellow->page->getContentHtml() ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?> 
```

As Giovanni explained you can also put everything in one file. You do you. 😄

**Comment by webNEXT-development** [comment, marked as answer]
Hello Giovanni Salmeri,

Thanks for your reply, I've edited the title. To be honest this was my first post on GitHub ever! so I'll pay close attention to the title when I ask another question.

Thank you for explaining my question, I put everything in 1 file.

Regards,
Marcel
